server/text_generation_server/models/__init__.py

```diff
@@ -227,6 +230,31 @@ def get_model(
                 dtype=dtype,
                 trust_remote_code=trust_remote_code,
             )
+        
+    elif model_type == "phi":
+        if FLASH_ATTENTION:
+            return FlashPhi(
+                model_id,
+                revision,
+                quantize=quantize,
+                dtype=dtype,
+                trust_remote_code=trust_remote_code,
+                use_medusa=use_medusa,
+            )
+        else:
+            raise NotImplementedError(FLASH_ATT_ERROR_MESSAGE.format("Phi"))
```

Comment:
Could we also load it with the Phi class based on CausalLM? Or is the layout different? 
I'm not too familiar with Phi.

Reply:
thanks for the feedback! 

yes, Phi can be loaded as a CausalLM with the `4.37.1` version of transformers. In my last PR I bumped the version and tested that Phi works on my local cpu